### ISSUE TITLE
"Accessibility" link in "Guides" menu directs to "References" section on MDN

### ISSUE THREAD

**Issue by glmvc** [mdn/yari#9309]
### Summary

The "Accessibility" link in the "Guides" menu links to the "References" section on MDN.
That's somehow confusing for users and unexpected behavior. Do you agree? Or is there another reason?

There is a "Accessibility" section in the learning area too where the submenu point should be linked to:
https://developer.mozilla.org/en-US/docs/Learn/Accessibility
Here is the line where the link is generated: https://github.com/mdn/yari/blob/main/client/src/ui/molecules/guides-menu/index.tsx#L59

---

Maybe the ["Accessibility" section](https://developer.mozilla.org/en-US/docs/Web/Accessibility) in the references area (where the current submenu link points to) should be added in the "References" menu?

But I don't know how the decision is made about which sections to include in the menus... For example, there are more sections in the learning area which _could_ be in the Guides menu, like e.g. "[Performance](https://developer.mozilla.org/en-US/docs/Learn/Performance)" or "[Web forms](https://developer.mozilla.org/en-US/docs/Learn/Forms)". But that's maybe a different topic?

### URL

main menu on every page

### Reproduction steps

1. Open the "Guides" submenu in the main navigation
2. Click on the "Accessibility" submenu link

### Expected behavior

To be directed to the "Accessibility" section in the **learning area**:
https://developer.mozilla.org/en-US/docs/Learn/Accessibility

### Actual behavior

Directed to the "Accessibility" section in the **references/docs area**:
https://developer.mozilla.org/en-US/docs/Web/Accessibility

### Device

Desktop

### Browser

Chrome

### Browser version

Stable

### Operating system

Mac OS

### Screenshot

![Screenshot 2023-07-16 at 11 18 20](https://github.com/mdn/yari/assets/95487764/fa88fe31-1dda-4e4f-a4f1-9ce867d16b86)

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/orgs/mdn/discussions/categories/platform).
